public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.test.tsx

```diff
@@ -0,0 +1,283 @@
+import { fireEvent, render, screen, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import 'whatwg-fetch';
+import { BootData } from '@grafana/data/src';
+import { selectors as e2eSelectors } from '@grafana/e2e-selectors/src';
+import { setEchoSrv } from '@grafana/runtime/src';
+import config from 'app/core/config';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { contextSrv } from 'app/core/services/context_srv';
+import { Echo } from 'app/core/services/echo/Echo';
+import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
+import { configureStore } from 'app/store/configureStore';
+
+import { ShareModal } from '../ShareModal';
+
+const server = setupServer(
+  rest.get('/api/dashboards/uid/:uId/public-config', (req, res, ctx) => {
```

Comment:
Is this mocked api specifically for RTK? Or would it still work if we're not using RTK?

Reply:
it would still work without using RTK Query. MSW is a library which intercepts the request in the network layer, so it's agnostic about how you make that request. 
However, it's recommended to use this library for RTK Query and I couldn't find another way of mocking the requests